filesystems/http/src/main/scala/cromwell/filesystems/http/HttpPathBuilder.scala

```diff
@@ -53,4 +53,15 @@ case class HttpPath(nioPath: NioPath) extends Path {
   override def pathAsString: String = nioPath.toString.replaceFirst("/", "//")
 
   override def pathWithoutScheme: String = pathAsString.replaceFirst("http[s]?://", "")
+
+  def fetchSize(implicit executionContext: ExecutionContext, actorSystem: ActorSystem): Future[Long] = {
+    Http().singleRequest(HttpRequest(uri = pathAsString, method = HttpMethods.HEAD)).map { response =>
+      response.discardEntityBytes()
+      val length = response.entity.contentLengthOption.getOrElse(0L)
```

Comment:
Can files that are 'touched' be size zero? Also I know Java has a method `orElseThow()` that I think Scala doesn't but you might be able to approximate it here as 
`
option.getOrElse(throw new RuntimeException("bad things"))
`

Reply:
It's slightly more subtle than it looks, because paths for which we get a 404 (or similar "I successfully made the request but there was no legitimate file there" code) also give us 0 length. So we want to throw an error when `response.entity.contentLengthOption` is either `Some(0)` or `None`.

As for whether we would want to support files that are legitimately 0... I'm not sure. I didn't want to make callers handle an optional return type, so I wanted to be sure we could either give a legitimate size or an error. I think this can be improved, though, so we return a 0 when it's really a 0 and return an error when it's not.